api/src/env.ts

```diff
@@ -380,37 +386,37 @@ export const getEnv = () => env;
  * When changes have been made during runtime, like in the CLI, we can refresh the env object with
  * the newly created variables
  */
-export function refreshEnv(): void {
+export async function refreshEnv(): Promise<void> {
 	env = {
 		...defaults,
 		...process.env,
-		...processConfiguration(),
+		...(await processConfiguration()),
 	};
 
 	process.env = env;
 
 	env = processValues(env);
 }
 
-function processConfiguration() {
+async function processConfiguration() {
 	const configPath = path.resolve(process.env['CONFIG_PATH'] || defaults['CONFIG_PATH']);
 
 	if (fs.existsSync(configPath) === false) return {};
 
-	const fileExt = path.extname(configPath).toLowerCase();
+	const fileExt = path.extname(configPath).toLowerCase().substring(1);
 
-	if (fileExt === '.js') {
-		const module = require(configPath);
-		const exported = module.default || module;
+	if (configFileExtensions.includes(fileExt)) {
```

Comment:
```suggestion
	if ((JAVASCRIPT_FILE_EXTS as readonly string[]).includes(fileExt)) {
```

Reply:
I had it inlined at first but that makes it a lot less readable and does not leave us the option to explain why we did this weird trick. If we're going inline i think i'd rather type cast the extension like 
```suggestion
	if (JAVASCRIPT_FILE_EXTS.includes(fileExt as any)) {
```

Reply:
Why `any`? `as readonly string[]` satisfies TS so we don't loose type safety and it's pretty self-explanatory, IMO.

Reply:
> Why any?

Because its the TS workaround that sticks closest to the approach you'd take if it wasnt a `readonly array` namely just `ARRAY.includes(value)`. Just because it has an `any` in there doesnt mean it loses type safety in any meaningful way in this case.

I strongly disagree about it being "self-explanatory" but it also doesnt really matter how we cast it the effect is the same, pick one 😄

Reply:
- IMO, `any` should always be the last resort, it's also considered bad practice by `typescript-eslint`, see [`no-explicit-any`](https://typescript-eslint.io/rules/no-explicit-any) - we've disabled this rule for now (with the hope that we can clean this up someday):
  https://github.com/directus/directus/blob/0c56c46fdf8dfd0a40ae04b3ec6270c42edf1077/.eslintrc.js#L50-L51
- You lose type safety because you wouldn't notice anymore if `fileExt` changes, e.g. from `string` to `object`
- `as readonly string[]` only works because it's compatible with the type of `JAVASCRIPT_FILE_EXTS` so we don't lose type safety with this. Try to change it to e.g. `readonly number[]`, TS would complain.
- `(JAVASCRIPT_FILE_EXTS as readonly string[]).includes(fileExt)` feels self-explanatory to me because it shows:
  - `JAVASCRIPT_FILE_EXTS` it's most likely a constant
  - that is compatible with `readonly string[]` so we see it's really a read only const with string compatible content
  - it can be compared to `fileExt` so fileExt is a string